nifi-extension-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/proto/ProtoUtils.java

```diff
@@ -46,7 +48,21 @@ public static DynamicMessage createMessage(Descriptors.Descriptor descriptor, Ma
            switch (field.getType()) {
            case MESSAGE:
                if (field.isRepeated()) {
-                   Collection collection = value.getClass().isArray() ? Arrays.asList((Object[]) value) : (Collection) value;
+                   Collection collection = null;
```

Comment:
As all possible code paths provide a value for this, it can be declared `final` and need not to be initialised.

And while you didn't choose the name `collection` here I'd like to use this change as an opportunity to provide a more meaningful name for this variable.

All code paths should lead to a collection of `Map<String, Object>` objects, thus we may as well define the type here instead of using a raw collection.

```suggestion
                   final Collection<Map<String, Object>> valueMaps;
```